File: benchmark/ruby/vector_dot.rb
a = []
b = []
for i in 1..500000
    a.push(i)
    b.push(i + 1)
end
sum = 0
start = Time.now.to_f * 1000
for i in 0..499999
    sum = sum + a[i] * b[i]
end
finish = Time.now.to_f * 1000
puts "Dot: #{sum}"
puts "Time: #{(finish - start).round}ms"

File: benchmark/nodejs/vector_dot.js
const a = [];
const b = [];
for (let i = 0; i < 500000; i = i + 1) {
    a.push(i + 1);
    b.push(i + 2);
}
let sum = 0n;
const start = Date.now();
for (let i = 0; i < 500000; i = i + 1) {
    sum = sum + BigInt(a[i]) * BigInt(b[i]);
}
const end = Date.now();
console.log(`Dot: ${sum}`);
console.log(`Time: ${end - start}ms`);
